include/grpcpp/security/tls_credentials_options.h

```diff
@@ -0,0 +1,281 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef GRPCPP_SECURITY_TLS_CREDENTIALS_OPTIONS_H
+#define GRPCPP_SECURITY_TLS_CREDENTIALS_OPTIONS_H
+
+#include <functional>
+#include <memory>
+#include <vector>
+
+#include <grpc/grpc_security.h>
+#include <grpc/support/log.h>
+#include <grpcpp/support/config.h>
+
+namespace grpc_impl {
+namespace experimental {
+
+/** TLS key materials config, wrapper for grpc_tls_key_materials_config. **/
+class TlsKeyMaterialsConfig {
+ public:
+  struct PemKeyCertPair {
+    ::grpc::string private_key;
+    ::grpc::string cert_chain;
+  };
+
+  /** Getters for member fields. **/
+  const ::grpc::string pem_root_certs() const { return pem_root_certs_; }
+  const ::std::vector<PemKeyCertPair>& pem_key_cert_pair_list() const {
+    return pem_key_cert_pair_list_;
+  }
+
+  /**Setter for member fields. **/
+  void set_key_materials(::grpc::string pem_root_certs,
+                         ::std::vector<PemKeyCertPair> pem_key_cert_pair_list);
+
+  /** Creates C struct for key materials. **/
+  grpc_tls_key_materials_config* c_key_materials() const;
```

Comment:
Do we need to expose this API? This seems to be internal details.

Reply:
It is an internal function. I could e.g. make it a private member function instead, and then make `TlsKeyMaterialsConfig` into a friend class of both `TlsCredentialReloadArg` and `TlsCredentialsOptions`. (Note that `c_key_materials` is called by the implementations of `c_credential_reload_arg` and `c_credentials_options`.)